app/src/test/java/com/willowtree/vocable/presets/PresetsViewModelTest.kt

```diff
@@ -114,6 +114,148 @@ class PresetsViewModelTest {
         )
     }
 
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun `selected category is hidden and next immediate category is shown`() = runTest(UnconfinedTestDispatcher()) {
+        fakeCategoriesUseCase._categories.update {
+            listOf(
+                Category.StoredCategory(
+                    categoryId = "1",
+                    localizedName = LocalesWithText(mapOf("en_US" to "category")),
+                    hidden = true,
+                    sortOrder = 0
+                ),
+                Category.StoredCategory(
+                    categoryId = "2",
+                    localizedName = LocalesWithText(mapOf("en_US" to "second category")),
+                    hidden = false,
+                    sortOrder = 1
+                )
+            )
+        }
+
+        val vm = createViewModel()
+
+        vm.onCategorySelected("1")
+
+        var category: Category? = null
+        val job = launch {
+            vm.selectedCategory.collect {
+                category = it
+            }
+        }
+        job.cancel()
```

Comment:
I _just_ brought in [Turbine](https://github.com/cashapp/turbine) to make this less obnoxious, could be fun to refactor this later and see how much nicer turbine makes things